src/Refactor/Entity/Product.php

```diff
@@ -0,0 +1,222 @@
+            $this->onlineTo = $to;
+        }
+    }
+
+    public function setSearchableFlags(?bool $availableFlag = null, ?bool $searchableFlag = null, ?bool $searchableIfUnavailableFlag = null): void
+    {
+        if ($availableFlag !== null) {
+            $this->availableFlag = $availableFlag;
+        }
+
+        if ($searchableFlag !== null) {
+            $this->searchableFlag = $searchableFlag;
+        }
+
+        if ($searchableIfUnavailableFlag !== null) {
+            $this->searchableIfUnavailableFlag = $searchableIfUnavailableFlag;
+        }
+    }
+
+    public function setTax(?float $tax): void
+    {
+        if ($tax === null) {
+            return;
+        }
+
+        if ($tax == 0) {
+            $tax = 'TAX_0';
+        } else {
+            $tax = number_format($tax, 2);
+
+            // Not sure why this needs to have two underscores.
+            if ($tax < 1) {
+                $tax = 'TAX__' . str_replace('0.', '', $tax);
+            } else {
+                $tax = 'TAX_' . str_replace('.', '_', $tax);
+            }
+        }
+
+        $this->tax = $tax;
+    }
+
+    public function setImages(array $images, string $imageViewType = 'large'): void
+    {
+        $this->imageViewType = $imageViewType;
+        $this->images        = $images;
+    }
+
+    public function setBrand(string $brand): void
+    {
+        $this->brand = $brand;
+    }
+
+    public function setSearchRank(int $searchRank = 3): void
+    {
+        $this->searchRank = $searchRank;
+    }
+
+    public function setSitemap(?float $sitemapPriority = null, bool $sitemapIncludedFlag = true, string $sitemapChangeFrequency = 'weekly'): void
+    {
+        if ($sitemapPriority !== null && $sitemapPriority > 1) {
+            throw new InvalidArgumentException('Sitemap priority must be 1.0 or less');
+        }
+
+        if ($sitemapPriority !== null) {
+            $this->sitemapPriority = number_format($sitemapPriority, 1);
+        }
+
+        $this->sitemapIncludedFlag    = $sitemapIncludedFlag;
+        $this->sitemapChangeFrequency = $sitemapChangeFrequency;
+    }
+
+    public function setPageAttributes(?string $pageTitle = null, ?string $pageDescription = null, ?string $pageKeywords = null, ?string $pageUrl = null): void
+    {
+        $this->pageAttributes = [
+            'page-title'       => $pageTitle,
+            'page-description' => $pageDescription,
+            'page-keywords'    => $pageKeywords,
+            'page-url'         => $pageUrl,
+        ];
+    }
+
+    public function addCustomAttribute(CustomAttribute $customAttribute): void
+    {
+        $this->customAttributes[$customAttribute->id] = $customAttribute;
+    }
+
+    public function setSharedVariationAttributes(array $sharedVariationAttributes): void
+    {
+        $this->sharedVariationAttributes = $sharedVariationAttributes;
+    }
+
+    public function setVariants(array $variants): void
+    {
+        foreach ($variants as $key => $value) {
```

Comment:
@WillJW I think it might be something to think about later, but for the time being I think we should concentrate on getting this development version tested.